src/main/java/com/gooddata/Authentication.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (C) 2004-2016, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+/**
+ * HttpClient factory
+ */
+public interface Authentication {
+
+    /**
+     * create http client for given host, port and protocol using given http client builder
+     * @param endpoint GoodData Platform's endpoint
+     * @param httpClientBuilder http client builder
+     * @return http client
+     */
+    HttpClient createHttpClient(final GoodDataEndpoint endpoint, final HttpClientBuilder httpClientBuilder);
```

Comment:
I am not sure what was the final agreement, but isn't it better to expose dependency on`ClientHttpRequestFactory` rather than on `HttpClient`?